packages/esm-form-entry-app/src/app/form-schema/form-schema.service.ts

```diff
@@ -1,59 +1,92 @@
 import { Injectable } from '@angular/core';
-import { ReplaySubject, Observable, Subject, of, forkJoin } from 'rxjs';
+import { ReplaySubject, Observable, Subject, of, forkJoin, zip, from } from 'rxjs';
 import { concat, first, map, take, tap } from 'rxjs/operators';
 
 import { FormResourceService } from '../openmrs-api/form-resource.service';
 import { FormSchemaCompiler } from '@openmrs/ngx-formentry';
 import { LocalStorageService } from '../local-storage/local-storage.service';
-import { FormSchema, Questions } from '../types';
+import { FormMetadataObject, FormSchema, FormSchemaAndTranslations, Questions } from '../types';
+import { TranslateService } from '@ngx-translate/core';
+import { merge } from 'lodash-es';
 
 @Injectable()
 export class FormSchemaService {
   constructor(
     private formsResourceService: FormResourceService,
     private localStorage: LocalStorageService,
     private formSchemaCompiler: FormSchemaCompiler,
+    private translateService: TranslateService,
   ) {}
 
-  public getFormSchemaByUuid(formUuid: string, cached: boolean = true): Observable<FormSchema> {
-    const cachedCompiledSchema = this.getCachedCompiledSchemaByUuid(formUuid);
-    if (cachedCompiledSchema && cached) {
-      return of(cachedCompiledSchema);
+  public getFormSchemaByUuid(
+    formUuid: string,
+    language: string = 'en',
+    cached: boolean = true,
+  ): Observable<FormSchema> {
+    if (cached) {
+      const cachedCompiledSchema = this.getCachedCompiledSchemaByUuid(formUuid, language);
```

Comment:
Caching now depends on language, not just the form UUID. For cached forms, the translations are pre-loaded into the cached schema. I haven't tested whether or not this works offline properly, though.